### DISCUSSION TITLE
How to create an entity with relation OneToMany, ManyToOne

### DISCUSSION THREAD

**Discussion by Maxttier** [mikro-orm/mikro-orm#6251] [General]
If I have that in entity

```

 @OneToMany(() => Profile, (profiles) => profiles.user, {
    cascade: [Cascade.ALL],
    lazy: true,
  })
  profiles?: Collection<Profile> = new Collection<Profile>(this)
```
and 

```
 @ManyToOne(() => User, { cascade: [Cascade.ALL], ref: true })
  user?: Ref<User> 
```

how I can create that entity and attach it to some user (without .attach())

like `repository.create({ id: 1, user: 1 })` or `repository.create({ id: 1, user_id: 1 })` or `repository.create({ id: 1, user: { id: 1 } })` or `repository.create({ id: 1, user: [{ id: 1 }] })`?

In other case (when is many) it work `repository.create({ id: 1, users: [1, 2, 3] })` 

```
     static create(entity) {
         const unwrapped = Reference.unwrapReference(entity);
         const ref = (0, wrap_1.helper)(entity).toReference();
                                                  ^
TypeError: undefined is not an object (evaluating 'wrap_1.helper(entity).toReference')
```

**Comment by Maxttier** [comment]
Without ref: true work

**Comment by B4nan** [comment]
It's not entirely clear to me what you are doing, please provide a [complete reproduction](https://github.com/mikro-orm/reproduction), it sounds like a bug on our end.

> like repository.create({ id: 1, user: 1 }) or repository.create({ id: 1, user_id: 1 }) or repository.create({ id: 1, user: { id: 1 } }) or repository.create({ id: 1, user: [{ id: 1 }] })?

Only the first one is correct, the others won't even compile I believe. The third one (`create({ id: 1, user: { id: 1 } })`) will work too, but the meaning is different, you create a new object with just the `id`, it will result in an insert query, while `user: 1` is considered an entity reference to an existing object, and won't produce an insert for that relation target.

**Comment by Maxttier** [comment]
I understand where is problem, I use:
```
  constructor(partial: Partial<EntiyName>) {
    super()
    Object.assign(this, partial)
  }
```

Bun in your example you use in constructor `ref(author);`:

```
constructor(author: Rel<User>, title: string, description = '', text = '') {
    super();
    this.author = ref(author);
    this.title = title;
    this.description = description;
    this.text = text;
    this.slug = convertToSlug(title);
  }

```

It work also without constructor.

**Reply by B4nan** [reply]
Yes, using `Object.assign(this, partial)` is indeed wrong, since it won't convert the foreign key to an entity reference (or as in your example, entity instance into the reference wrapper), it will just assign things. There is also `assign()` ORM helper doing this, which will do it for you, although I think it won't be able to create unmanaged entity references from FKs (something I keep forgetting to fix, lol), but if you are passing entity instance in there, it should work fine.

https://mikro-orm.io/docs/entity-helper
